pkg/user/service.go

```diff
@@ -124,3 +130,12 @@ func (s service) FindOrCreate(email string, password string) (*model.User, error
 
 	return s.repository.findOrCreate(user)
 }
+
+func (s service) Delete(id uint) error {
+	_, err := s.repository.findById(id)
```

Comment:
We are making 2 separate operations on the DB while one would do it. For example
 https://stackoverflow.com/questions/67154864/how-to-handle-gorm-error-at-delete-function